custom_components/weenect/sensor.py

```diff
@@ -124,7 +198,23 @@ class WeenectSensor(WeenectEntity, SensorEntity):
     def native_value(self) -> StateType | datetime:
         """Return the state of the resources if it has been received yet."""
         if self.id in self.coordinator.data:
-            return self.coordinator.data[self.id][self.entity_description.key]
+            if self.entity_description.key == "call_available":
+                if (
+                    "call_usage" in self.coordinator.data[self.id]
+                    and "call_max_threshold" in self.coordinator.data[self.id]
+                ):
+                    return (
+                        self.coordinator.data[self.id]["call_max_threshold"]
+                        - self.coordinator.data[self.id]["call_usage"]
+                    )
+                else:
+                    return None
+            value = self.coordinator.data[self.id][self.entity_description.key]
+            if self.entity_description.key == "expiration_date":
+                value = datetime.strptime(value, "%Y-%m-%dT%H:%M:%S.%f").replace(
+                    tzinfo=timezone.utc
+                )
+            return value
```

Comment:
Since this repeats in the UserSensor, is this something we can generalize for device_class=SensorDeviceClass.TIMESTAMP?